src/Functions/DateTimeTransforms.h

```diff
@@ -57,25 +60,51 @@ struct ZeroTransform
     static UInt16 execute(UInt16, const DateLUTImpl &) { return 0; }
 };
 
+template <DateTimeOverflowMode date_time_overflow_mode = default_date_time_overflow_mode>
 struct ToDateImpl
 {
     static constexpr auto name = "toDate";
 
     static UInt16 execute(const DecimalUtils::DecimalComponents<DateTime64> & t, const DateLUTImpl & time_zone)
     {
-        return static_cast<UInt16>(time_zone.toDayNum(t.whole));
+        return execute(t.whole, time_zone);
     }
+
     static UInt16 execute(Int64 t, const DateLUTImpl & time_zone)
     {
-        return UInt16(time_zone.toDayNum(t));
+        if constexpr (date_time_overflow_mode == DateTimeOverflowMode::SATURATE)
+        {
+            if (t < 0)
+                t = 0;
+            else if (t > 5662310399)    // 2149-06-06 23:59:59 UTC
+                t = 5662310399;
+        }
+        else if constexpr (date_time_overflow_mode == DateTimeOverflowMode::THROW)
+        {
+            if (unlikely(t < 0 || t > 5662310399))
+                throw Exception(ErrorCodes::DATE_TIME_OVERFLOW, "Value {} is out of bounds of type Date", t);
+        }
+        return static_cast<UInt16>(time_zone.toDayNum(t));
```

Comment:
Let's move this into an `else` branch. Same for l. 107.

Reply:
Here it is not an "else" thing. The checks in the lined above are optional, and if the number is fine, l.87 is executed. If mode is `THROW` and there is no overflow, this line if executed.
This is to avoid duplication of `return static_cast<UInt16>(time_zone.toDayNum(...));` in all branches.

Reply:
You are right, tnx.